tools/txg/TXG.hs

```diff
@@ -100,197 +100,168 @@ generateDelay = do
-  (timeTaken, !_action) <- measureDiffTime go
-  when mtime
-    . withConsoleLogger Info
-    . logg Info
-    $ "Loading supplied contracts took: " <> sshow timeTaken
-  where
-    go :: IO ()
-    go = do
-      TXGConfig _ _ ce v <- mkTXGConfig Nothing config host
-      forM_ (_nodeChainIds config) $ \cid -> do
-        let !meta = Sim.makeMeta cid
-        ts <- testSomeKeyPairs
-        contracts <- traverse (\f -> f meta ts) contractLoaders
-        pollresponse <- runExceptT $ do
-          rkeys <- ExceptT $ runClientM (send v cid $ SubmitBatch contracts) ce
-          ExceptT $ runClientM (poll v cid . Poll $ _rkRequestKeys rkeys) ce
-        withConsoleLogger Info . logg Info $ sshow pollresponse
+loadContracts :: Args -> HostAddress -> [ContractLoader] -> IO ()
+loadContracts config host contractLoaders = do
+  TXGConfig _ _ ce v _ <- mkTXGConfig Nothing config host
+  forM_ (_nodeChainIds config) $ \cid -> do
+    let !meta = Sim.makeMeta cid
+    ts <- testSomeKeyPairs
+    contracts <- traverse (\f -> f meta ts) contractLoaders
+    pollresponse <- runExceptT $ do
+      rkeys <- ExceptT $ runClientM (send v cid $ SubmitBatch contracts) ce
+      ExceptT $ runClientM (poll v cid . Poll $ _rkRequestKeys rkeys) ce
+    withConsoleLogger Info . logg Info $ sshow pollresponse
 
 sendTransactions
-  :: MeasureTime
-  -> ScriptConfig
+  :: Args
   -> HostAddress
   -> TVar TXCount
   -> TimingDistribution
   -> LoggerT SomeLogMessage IO ()
-sendTransactions measure config host tv distribution = do
-  cfg@(TXGConfig _ _ ce v) <- liftIO $ mkTXGConfig (Just distribution) config host
+sendTransactions config host tv distribution = do
+  cfg@(TXGConfig _ _ ce v _) <- liftIO $ mkTXGConfig (Just distribution) config host
 
-  accountMap <- fmap (M.fromList . toList) . forM (_nodeChainIds config) $ \cid -> do
+  let chains = maybe (versionChains $ _nodeVersion config) NES.fromList
+               . NEL.nonEmpty
+               $ _nodeChainIds config
+
+  accountMap <- fmap (M.fromList . toList) . forM chains $ \cid -> do
     let !meta = Sim.makeMeta cid
     (paymentKS, paymentAcc) <- liftIO $ unzip <$> Sim.createPaymentsAccounts meta
     (coinKS, coinAcc) <- liftIO $ unzip <$> Sim.createCoinAccounts meta
     pollresponse <- liftIO . runExceptT $ do
       rkeys <- ExceptT $ runClientM (send v cid . SubmitBatch $ paymentAcc ++ coinAcc) ce
       ExceptT $ runClientM (poll v cid . Poll $ _rkRequestKeys rkeys) ce
-    logg Info $ toLogMessage (sshow pollresponse :: Text)
-    logg Info $ toLogMessage ("Transactions are being generated" :: Text)
+    case pollresponse of
+      Left e -> logg Error $ toLogMessage (sshow e :: Text)
+      Right _ -> pure ()
     let accounts = buildGenAccountsKeysets Sim.accountNames paymentKS coinKS
     pure (cid, accounts)
 
+  logg Info $ toLogMessage ("Real Transactions: Transactions are being generated" :: Text)
+
   -- Set up values for running the effect stack.
   gen <- liftIO createSystemRandom
```

Comment:
What about adding a config option for a seed create all generators from that, so that tests can be replayed?

Reply:
Reasonable.

Reply:
I will leave this as "future work".